mifosng-provider/src/integrationTest/java/org/mifosplatform/integrationtests/common/SearchHelper.java

```diff
@@ -0,0 +1,189 @@
+package org.mifosplatform.integrationtests.common;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.path.json.JsonPath.from;
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import org.junit.Test;
+
+import com.jayway.restassured.specification.RequestSpecification;
+import com.jayway.restassured.specification.ResponseSpecification;
+
+public class SearchHelper {
+	
+	private final RequestSpecification requestSpec;
+    private final ResponseSpecification responseSpec;
+    
+    private static final String SEARCH_CLIENT_URL = "/mifosng-provider/api/v1/search?"+ Utils.TENANT_IDENTIFIER;
+    private static final String GET_CLIENT_URL = "/mifosng-provider/api/v1/clients/"  + Utils.TENANT_IDENTIFIER;
+    static HashMap<String, Object> detail=new HashMap<String,Object>(); 
+    
+    public SearchHelper(final RequestSpecification requestSpec, final ResponseSpecification responseSpec) {
+        this.requestSpec = requestSpec;
+        this.responseSpec = responseSpec;
+    }
+   
+    public static Object exactClientSearch (final RequestSpecification requestSpec, final ResponseSpecification responseSpec, final String searchParameter,
```

Comment:
reduce the number of functions.Have one function each for client,loan,group,savings etc and pass the value true/false by passing parameter exactMatch .Define all URLs globally.